src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -1837,137 +1640,112 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
     } break;
     case DataGlobalConstants::CallIndicator::EndDay: {
         // average some of the zone sequences to reduce peakiness
-        for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
+        for (int CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
             if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+            auto &calcZoneSizing(state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum));
             state.dataZoneEquipmentManager->AvgData = 0.0;
-            MovingAvg(state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolFlowSeq,
-                      state.dataZoneEquipmentManager->NumOfTimeStepInDay,
-                      state.dataSize->NumTimeStepsInAvg,
-                      state.dataZoneEquipmentManager->AvgData);
-            state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolFlowSeq = state.dataZoneEquipmentManager->AvgData;
-        }
-        for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
-            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+            General::MovingAvg(calcZoneSizing.CoolFlowSeq,
+                               state.dataZoneEquipmentManager->NumOfTimeStepInDay,
+                               state.dataSize->NumTimeStepsInAvg,
+                               state.dataZoneEquipmentManager->AvgData);
+            calcZoneSizing.CoolFlowSeq = state.dataZoneEquipmentManager->AvgData;
```

Comment:
Initial loop compression effort. Each array had their own loop, moved into a single loop since all were looping through number of zones.